src/main/deltachat.js

```diff
@@ -6,24 +6,22 @@ const config = require('../config')
 
 class ChatMessage {
   constructor (messageId, dc) {
-    var msg = dc.getMessage(messageId)
-    this.fromId = msg && msg.getFromId()
-    if (!this.fromId) return
-
-    this.msg = msg
-    this.messageId = messageId
-    this.isMe = this.fromId === 1
-    this.contact = dc.getContact(this.fromId)
+    this._messageId = messageId
+    this._dc = dc
   }
 
   toJson () {
+    var msg = this._dc.getMessage(this._messageId)
```

Comment:
Since we call `dc.getMessage()` in `toJson()` we will always get the correct state when `render()` is called.